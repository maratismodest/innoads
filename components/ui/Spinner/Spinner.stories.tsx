import type { Meta, StoryObj } from '@storybook/react';
import Spinner from './index';

const meta = {
  title: 'Components/Spinner',
  component: Spinner,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  args: {
    children: 'Click me',
  },
  argTypes: {
    // variant: {
    //   options: ['primary', 'secondary'],
    //   control: { type: 'radio' },
    // },
    // size: {
    //   options: ['medium', 'small'],
    //   control: { type: 'radio' },
    // },
  },
} satisfies Meta<typeof Spinner>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};

// export const Primary: Story = {
//   args: {
//     variant: 'primary',
//     size: 'small',
//   },
// };
//
// export const Secondary: Story = {
//   args: {
//     variant: 'secondary',
//     size: 'small',
//   },
// };
