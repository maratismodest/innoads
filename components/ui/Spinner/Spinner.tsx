import { clsx } from 'clsx';
import React, { memo } from 'react';

interface Props extends React.ComponentPropsWithoutRef<'div'> {}

const Spinner = ({ className }: Props) => {
  return (
    <div className={clsx('flex items-center justify-center p-8', className)}>
      <div className="h-16 w-16 animate-spin rounded-full border-b-2 border-blue"></div>
    </div>
  );
};

export default memo(Spinner);
