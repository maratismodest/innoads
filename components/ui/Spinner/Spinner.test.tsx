import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import React from 'react';

import Spinner from './Spinner';

describe('Spinner Component', () => {
  it('renders with default props', () => {
    const { container } = render(<Spinner />);
    const spinnerElement = container.querySelector('.animate-spin');
    expect(spinnerElement).toBeInTheDocument();
    expect(spinnerElement).toHaveClass('h-16');
    expect(spinnerElement).toHaveClass('w-16');
    expect(spinnerElement).toHaveClass('border-b-2');
    expect(spinnerElement).toHaveClass('border-blue');
  });

  it('renders with custom className', () => {
    const { container } = render(<Spinner className="custom-class" />);
    const spinnerContainer = container.querySelector('.flex');
    expect(spinnerContainer).toHaveClass('custom-class');
  });

  it('matches snapshot', () => {
    const { container } = render(<Spinner />);
    expect(container).toMatchSnapshot();
  });
});
