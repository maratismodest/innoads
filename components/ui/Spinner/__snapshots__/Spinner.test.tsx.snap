// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Component matches snapshot 1`] = `
<div>
  <div
    class="flex items-center justify-center p-8"
  >
    <div
      class="h-16 w-16 animate-spin rounded-full border-b-2 border-blue"
    />
  </div>
</div>
`;
